src/storage/double_write_buffer.cpp

```diff
@@ -204,6 +204,7 @@ struct flush_volume_info
   int vdes;			/* The volume descriptor. */
   volatile int num_pages;	/* The number of pages to flush in volume. */
   volatile bool all_pages_written;	/* True, if all pages written. */
+  volatile bool metadata;	/* Include metadata when syncing */
```

Comment:
metadata로 표현된 부분에 일괄 적용 검토; 주석은 부차적인 것으로 변수에 명확한 의미를 담는 것이 중요합니다.
```suggestion
  volatile bool sync_metadata;	/* Include metadata when syncing */
```

Reply:
flush_volume_info는 기본적으로 flush 대상이 되는 볼륨에 대한 정보를 담고 있습니다. 따라서 변수 이름도 `vdes`, `num_pages`의 형태이고, 이에 따라 metadata의 여부를 나타내는 변수 `bool metadata`를 추가하였습니다.
만약 `metadata`를 `sync_metadata`로 바꿔야 한다면 다른 네이밍도 `vdes`에서 `target_vdes`, `include_num_pages` 처럼 바뀌어야 할 것 같습니다.
일관성 측면에서는 지금의 네이밍을 유지하는 게 더 좋지 않을까요?

Reply:
저도 코드를 읽어 보다보니, 다음과 같은 부분에서 변수 이름 표현에 의해 그 의미를 헷갈릴 수 있을 것 같습니다.
```
current_flush_volume_info = dwb_add_volume_to_block_flush_area (thread_p, block, last_written_vol_fd, metadata);
```

Reply:
음. 넘겨주는 부분에서는 조금 더 명확한 이름을 쓰도록 수정하겠습니다.
함수의 경우에도 sync를 요 하는 부분은 `metadata`로 유지하고, 아닌 경우에는 `ensure_metadata`로 변경하겠습니다.
`sync_...`의 경우에는 함수가 직접 sync하는 느낌이 있고 metadata만 sync하는 느낌이 있어 피하려고 합니다.

Reply:
저는 좀 다른 관점입니다. metadata라는 변수명을 보았을 때 의미를 파악하기 위해 사용처를 확인해야 합니다(번거로움). sync_metadata를 제한드린 이유도 이런 부분을 최소화하기 위해서 입니다. bool metadata인 경우 이 boolean 변수의 true, false 의미가 무엇인지 유추하기 어렵습니다(metadata를 어떻게 하라는 것이지?). 또한 if 조건 등에서 조건 비교 시 가독성(의미 파악 관점) 역시 떨어집니다. bool sync_metadata인 경우 true -> sync metadata, false -> no sync metadata로 의미가 명확해지기 때문에 위 문제들이 해소될 수 있습니다.

추가적으로, 관련 변수는 모두 통일하여 동일한 변수명을 사용하는게 추후 위에서 설명드린 문제를 최소화할 수 있을 것 같아서 모두 일관되게 수정하시는 것을 제안합니다.

Reply:
현재 `flush_volume_info` 내부만 `metadata`로 사용하고, 나머지는 전부 `ensure_metadata`로 통일되어 있는데 부족할까요? `flush_volume_info`는 선언 자체가 sync 대상들이므로 충분하다고 생각합니다.

Reply:
NIT:

저는 모두 반영하는게 적절할 것 같습니다. 검토 시 dwb_add_volume_to_block_flush_area() 인자에도 반영 고려해 주세요.